webui/src/pages/auth/login.jsx

```diff
@@ -23,18 +23,19 @@ const LoginForm = () => {
                 <Card className="login-widget">
                     <Card.Header>Login</Card.Header>
                     <Card.Body>
-                        <Form onSubmit={async (e) => {
+                        <Form onSubmit={(e) => {
                             e.preventDefault()
-                            try {
-                                await auth.login(e.target.username.value, e.target.password.value);
-                                setLoginError(null);
-                                const nextOrRoot = next && next !== router.route ? next : '/';
-                                router.push(nextOrRoot);
-                            } catch(err) {
-                                setLoginError(err);
-                            }
-                        }}>
+                            auth
+                                .login(e.target.username.value, e.target.password.value)
+                                .then(user => {
+                                    setLoginError(null);
+                                    return router.push((!!next) ? next : '/');
+                                })
+                                .catch(err => {
+                                    setLoginError(err);
+                                })
```

Comment:
I think this may be more readable as an async function (cannot put it as a suggestion because GitHub).

Something like this (untested, of course):
```js
async (e) => {
  e.preventDefault();
  try {
    const _user = await auth.login(e.target.username.value, e.target.password.value);
    setLoginError(null);
    return router.push((!!next) ? next : '/');
    } catch (e) {
      setLoginError(e);
    }
}
```

Reply:
:) I revered this change because I didn't wanted to add code style changes to the PR - but I also found the async syntax better to read.